presto-hive/src/main/java/io/prestosql/plugin/hive/util/HiveFileIterator.java

```diff
@@ -101,7 +110,17 @@ protected LocatedFileStatus computeNext()
             if (paths.isEmpty()) {
                 return endOfData();
             }
-            remoteIterator = getLocatedFileStatusRemoteIterator(paths.removeFirst());
+
+            Path path = paths.removeFirst();
+            String payload = null;
+            if (tracer.isEnabled()) {
+                Map<String, Object> map = new HashMap<>();
+                map.put("path", path.toString());
+                payload = jsonCodec.toJson(map);
+                tracer.emitEvent(ConnectorTracerActionType.LIST_FILE_STATES_START, payload);
```

Comment:
I think this can be compacted: 
```
if (tracer.isEnabled()) {
    tracer.event(LIST_FILE_STATS_START, toJson(ImmutableMap.of("path", path));
}
```

also applicable to other emissions